lib/connection.js

```diff
@@ -102,7 +102,20 @@ Connection.prototype.connect = function (port, host) {
       key: self.ssl.key,
       passphrase: self.ssl.passphrase,
       cert: self.ssl.cert,
+      ciphers: self.ssl.ciphers,
+      sigalgs: self.ssl.sigalgs,
+      clientCertEngine: self.ssl.clientCertEngine,
+      crl: self.ssl.crl,
+      dhparam: self.ssl.dhparam,
+      ecdhCurve: self.ssl.ecdhCurve,
+      honorCipherOrder: self.ssl.honorCipherOrder,
+      privateKeyEngine: self.ssl.privateKeyEngine,
+      privateKeyIdentifier: self.ssl.privateKeyIdentifier,
+      maxVersion: self.ssl.maxVersion,
+      minVersion: self.ssl.minVersion,
+      minDHSize: self.ssl.minDHSize,
       secureOptions: self.ssl.secureOptions,
+      ALPNProtocols: self.ssl.ALPNProtocols,
```

Comment:
what if we just did 

```js
tls.connect(Object.assign({
  // all the non-ssl options here
}, self.ssl))
```

Reply:
Works for me! I'll push an update shortly